source/core/src/main/com/csse3200/game/areas/terrain/CropTileComponent.java

```diff
@@ -0,0 +1,111 @@
+package com.csse3200.game.areas.terrain;
+
+import com.badlogic.gdx.math.MathUtils;
+import com.csse3200.game.components.Component;
+import com.csse3200.game.services.ServiceLocator;
+
+/**
+ * Component which stores information about plots of land on which crops and plants can grow.
+ * Listens to a variety of trigger-able events to update state of the tile, including its water
+ * content and whether it is fertilised.
+ */
+public class CropTileComponent extends Component {
+
+    private final float waterDecreaseRate = 0.05f;
+
+    private final float idealWaterFalloffSharpness = 2.0f;
+    private final float idealWaterFalloffTolerance = 1.2f;
+    private final float waterDamageThreshold = 0.1f;
+
+    private float waterContent;
+    private final float soilQuality;
+    private boolean isFertilised;
+    private boolean isOccupied;
+
+    /**
+     * Creates a new crop tile component, with a specified initial water content, and an initial soil quality (higher values indicate higher soil
+     * quality, meaning better growth rate).
+     *
+     * @param initialWaterContent The initial amount of water in the crop tile. Value should be
+     *                            between 0 and 2 inclusive - 0 indicates tile is completely dry,
+     *                            and 2 indicates tile is flooded, water content of 1 leads to ideal
+     *                            growth rate for most plants
+     * @param soilQuality The initial soil quality of the crop tile. Higher soil quality increases
+     *                    the calculated growth rate. Should be greater than 0.
+     */
+    public CropTileComponent(float initialWaterContent, float soilQuality) {
+        this.waterContent = initialWaterContent;
+        this.soilQuality = soilQuality;
+        this.isFertilised = false;
+        this.isOccupied = false;
+    }
+
+    /**
+     * Adds player-triggered events listeners to the entity.
+     */
+    @Override
+    public void create() {
+        entity.getEvents().addListener("water", this::waterTile);
+        entity.getEvents().addListener("fertilise", this::fertiliseTile);
+        entity.getEvents().addListener("plant", this::plantCrop);
+        entity.getEvents().addListener("harvest", this::harvestCrop);
+    }
+
+    /**
+     * Decreases water content in the tile by a constant amount.
+     */
+    @Override
+    public void update() {
+        waterContent -= waterDecreaseRate * ServiceLocator.getTimeSource().getDeltaTime();
+        waterContent = waterContent < 0 ? 0 : (waterContent > 2 ? 2 : waterContent);
+    }
+
+    private void waterTile(float amount) {
+        waterContent += amount;
+    }
+
+    private void fertiliseTile() {
+        isFertilised = true;
+    }
+
+    private void plantCrop() {
+        if (!isOccupied) {
+            isOccupied = true;
+        }
+    }
```

Comment:
Are we handling the case of the tile already being occupied?

Reply:
Yes, if it is occupied it doesn't do anything.